components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -266,6 +271,41 @@ private void validateClaimValueForRegex(String claimURI, String claimValue, Stri
         }
     }
 
+    /**
+     * Validate attribute values against length limits.
+     *
+     * @param claimURI      Claim URI.
+     * @param value         Claim value.
+     * @param tenantDomain  Tenant domain name..
+     * @throws UserStoreClientException If an error occurred in validating claim.
+     */
+    private void validateLength(String claimURI, String value, String tenantDomain) throws UserStoreClientException {
+
+        if (StringUtils.isBlank(claimURI)) {
+            if (log.isDebugEnabled()) {
+                log.debug("The claim URI is empty.");
+            }
+            return;
+        }
+        Map<String, String> claimProperties = getClaimProperties(tenantDomain, claimURI);
+        if (MapUtils.isEmpty(claimProperties)) {
+            return;
+        }
+        String minLength = claimProperties.get(MIN_LENGTH);
+        String maxLength = claimProperties.get(MAX_LENGTH);
+        boolean required = Boolean.parseBoolean(claimProperties.get(REQUIRED));
+
+        if (!required && StringUtils.isBlank(value)) {
+            return;
+        }
+        if ((StringUtils.isNotEmpty(minLength) && Integer.parseInt(minLength) > value.length()) ||
+                (StringUtils.isNotEmpty(maxLength) && Integer.parseInt(maxLength) < value.length())) {
+            throw new UserStoreClientException(String.format(ERROR_CODE_LENGTH_VIOLATION.getDescription(),
+                    claimProperties.get(PROP_DISPLAYNAME), StringUtils.isNotEmpty(minLength) ? minLength : 0,
+                    StringUtils.isNotEmpty(maxLength) ? maxLength : 1024), ERROR_CODE_LENGTH_VIOLATION.getCode());
```

Comment:
What is the purpose of giving the threshold values in an exception?

Reply:
If any of min or max length is configured, we do the validation and return the error. If either of them is not configured, we display the threshhold values.

Reply:
ok